WebKit/WebKit
# XSLTProcessor fails when xsl calls exsl:node-set() on empty variable

#### 6cd705fc515c22b3be8ae770c410777b83bad088
<pre>
XSLTProcessor fails when xsl calls exsl:node-set() on empty variable
<a href="https://bugs.webkit.org/show_bug.cgi?id=253857">https://bugs.webkit.org/show_bug.cgi?id=253857</a>
&lt;rdar://103622929&gt;

Reviewed by Alex Christensen.

Merge fix for Chromium Issue 689977:
<a href="https://chromium.googlesource.com/chromium/src.git/+/99ead7d1564d35a70799b7ee4c3821053fb3985c">https://chromium.googlesource.com/chromium/src.git/+/99ead7d1564d35a70799b7ee4c3821053fb3985c</a>

Tests:
    fast/xsl/exslt-node-set.xml
    fast/xsl/xslt-node-set-empty.html

* LayoutTests/fast/xsl/xslt-node-set-empty-expected.txt: Add.
* LayoutTests/fast/xsl/xslt-node-set-empty.html: Add.
* Source/WebCore/xml/SoftLinkLibxslt.cpp:
* Source/WebCore/xml/SoftLinkLibxslt.h:
- Add soft-linking for newly called libxslt functions.
* Source/WebCore/xml/XSLTExtensions.cpp:
(xsltTransformErrorTrampoline): Add.
- Provide workaround for soft-linking xsltTransformError()
  with varargs.
(WebCore::exsltNodeSetFunction):
- Update to match logic in libxslt v1.1.35.

Originally-landed-as: 259548.430@safari-7615-branch (06af3d226e2b). rdar://104290899
</pre>
----------------------------------------------------------------------
#### 97f3965e6196c24dd33f290ba47789475145fe54
<pre>
VideoFrame constructor should throw on tainted input
<a href="https://bugs.webkit.org/show_bug.cgi?id=253828">https://bugs.webkit.org/show_bug.cgi?id=253828</a>
rdar://problem/106643249

Reviewed by Jean-Yves Avenard.

Add a CORS check in VideoFrame constructor code paths.

Covered by added tests.

* LayoutTests/TestExpectations:
* LayoutTests/http/wpt/webcodecs/green.png: Added.
* LayoutTests/http/wpt/webcodecs/image.py: Added.
(main):
* LayoutTests/http/wpt/webcodecs/webcodecs-crossOrigin.sub-expected.txt: Added.
* LayoutTests/http/wpt/webcodecs/webcodecs-crossOrigin.sub.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webcodecs/videoFrame-canvasImageSource-crossOrigin.sub-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webcodecs/videoFrame-canvasImageSource-crossOrigin.sub.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webcodecs/videoFrame-canvasImageSource-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webcodecs/videoFrame-canvasImageSource.html:
* Source/WebCore/Modules/webcodecs/WebCodecsVideoFrame.cpp:
(WebCore::isCachedImageTaintsOrigin):
(WebCore::checkImageUsability):
(WebCore::WebCodecsVideoFrame::create):
* Source/WebCore/css/CSSImageValue.h:
* Source/WebCore/css/typedom/CSSStyleImageValue.h:

Originally-landed-as: 259548.440@safari-7615-branch (2ef6b76e1902). rdar://104290899
</pre>
https://github.com/WebKit/WebKit/commit/6cd705fc515c22b3be8ae770c410777b83bad088

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/7244 "1 style error") | [❌ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/7492 "Failed to compile WebKit") | [❌ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/7668 "Failed to compile WebKit") | [❌ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/8864 "Failed to compile WebKit") | [❌ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/7454 "Failed to compile WebKit") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/7252 "Passed tests") | [❌ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/8825 "Failed to compile WebKit") | [❌ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/7421 "Failed to compile WebKit") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/5/builds/8864 "Failed to compile WebKit") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/7371 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/23/builds/8825 "Failed to compile WebKit") | [❌ 🧪 api-mac](https://ews-build.webkit.org/#/builders/14/builds/7668 "Failed to compile WebKit") | [❌ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/8970 "Failed to compile WebKit") | 
| | [❌ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/23/builds/8825 "Failed to compile WebKit") | [❌ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/14/builds/7668 "Failed to compile WebKit") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/2/builds/8970 "Failed to compile WebKit") | 
| | [❌ 🧪 api-ios](https://ews-build.webkit.org/#/builders/23/builds/8825 "Failed to compile WebKit") | [❌ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/14/builds/7668 "Failed to compile WebKit") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/2/builds/8970 "Failed to compile WebKit") | 
| | [❌ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/7176 "Failed to compile WebKit") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/16/builds/7421 "Failed to compile WebKit") | | 
| | [❌ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/6534 "Failed to compile WebKit") | [❌ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/14/builds/7668 "Failed to compile WebKit") | | 
| | [❌ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/10734 "Failed to compile WebKit") | | | 
| | [❌ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/6914 "Failed to compile WebKit") | | |
